File: ECommerce/ECommerce.Domain/Entities/Pedido.cs
using System;
using System.Collections.Generic;

namespace ECommerce.Domain.Entities
{
    public class Pedido
    {
        public int IdPedido { get; set; }
        public int IdStatus { get; set; }
        public virtual StatusPedido Status { get; set; }
        public decimal ValorPedido { get; set; }
        public DateTime DataPrevisaoEntrega { get; set; }
        public virtual List<ItemPedido> ItensPedido { get; set; }
    }
}

File: ECommerce/ECommerce.Presentation/ClientApp/src/app/Entitiys/Pedido.ts
import { StatusPedido } from "./StatusPedido";
import { ItemPedido } from "./ItemPedido";

export class Pedido {
    IdPedido: number
    IdStatus: number
    Status: StatusPedido;
    ValorPedido: number;
    DataPrevisaoEntrega: Date;
    ItensPedidos: ItemPedido;
}